### ISSUE TITLE
milvus.client.exceptions.NotConnectError: Fail connecting to server on localhost:19530

### ISSUE THREAD

**Issue by beyondguo** [milvus-io/milvus#1664]
**What is your question?**
Windows，python3.6，pymilvus 0.2.8

code:
```
from milvus import Milvus, IndexType, MetricType, Status
milvus = Milvus()
milvus.connect(host='localhost',port='19530')
```

error:
Traceback (most recent call last):
  File "E:\learningsoft\anadonda\lib\site-packages\milvus\client\grpc_handler.py", line 212, in connect
    grpc.channel_ready_future(self._channel).result(timeout=timeout)
  File "E:\learningsoft\anadonda\lib\site-packages\grpc\_utilities.py", line 140, in result
    self._block(timeout)
  File "E:\learningsoft\anadonda\lib\site-packages\grpc\_utilities.py", line 86, in _block
    raise grpc.FutureTimeoutError()
grpc.FutureTimeoutError

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "vector_search.py", line 5, in <module>
    milvus.connect(host='localhost',port='19530')
  File "E:\learningsoft\anadonda\lib\site-packages\milvus\client\stub.py", line 76, in connect
    return self._handler.connect(host, port, uri, timeout)
  File "E:\learningsoft\anadonda\lib\site-packages\milvus\client\grpc_handler.py", line 216, in connect
    raise NotConnectError('Fail connecting to server on {}. Timeout'.format(self._uri))
milvus.client.exceptions.NotConnectError: Fail connecting to server on localhost:19530. Timeout

**Comment by BossZou** [comment]
Hello, please make sure your server is on when you try to connect it. The trace log show that your client cannot find server.

**Comment by beyondguo** [comment]
Need help....
I tried it on a Linux conputer, but the same error.

**Comment by BossZou** [comment]
How do you launch you server? Use a docker container or start from source code?
And may you provide the server config file here? @beyondguo

**Comment by BossZou** [comment]
@beyondguo Did you solved this issue? If you have any problem, please let me know.

**Comment by seth814** [comment]
I think he is just not starting the server first since this is not obvious when you read the pymilvus docs. I started with pymilvus so run into this issue as well.

@beyondguo You should go here: https://www.milvus.io/docs/v0.7.0/guides/get_started/install_milvus/install_milvus.md

And use docker to run the server. Install for cpu or gpu. I followed the instructions for a gpu server and was able to launch with docker and connect on port 19530 using pymilvus.

**Comment by BossZou** [comment]
@beyondguo Did you solved this issue? If you have any problem, please let me know.

**Comment by BossZou** [comment]
No response for a long time, close it.

**Comment by rajeevkri** [comment]
I am also facing the same issue my Milvus pod is running as service at port 19530.

milv-milvus                ClusterIP   10.103.99.16     <none>        19530/TCP,9091/TCP

==============

But when trying to connect

`>>> from pymilvus import connections
>>> connections.connect("default", host='localhost', port='19530')
Retry to connect server localhost:19530 failed.
Traceback (most recent call last):
  File "/home/rajeevkrishna/.local/lib/python3.8/site-packages/pymilvus/client/grpc_handler.py", line 303, in ping
    ft.result(timeout=timeout)
  File "/home/rajeevkrishna/.local/lib/python3.8/site-packages/grpc/_utilities.py", line 140, in result
    self._block(timeout)
  File "/home/rajeevkrishna/.local/lib/python3.8/site-packages/grpc/_utilities.py", line 86, in _block
    raise grpc.FutureTimeoutError()
grpc.FutureTimeoutError
`

**Comment by jmugan** [comment]
Ugh, me too

**Comment by yiwangdr** [comment]
If you hit this issue while connecting to milvus from minikube, you might need
`kubectl port-forward service/my-release-milvus 19530:19530`.
^ replace `my-release-milvus` with your milvus service name.

**Comment by chtang-wish** [comment]
@yiwangdr is possible to add them into https://raw.githubusercontent.com/milvus-io/milvus-operator/main/config/samples/milvus_default.yaml?

**Comment by Kheem-Dh** [comment]
MilvusException(Status.CONNECT_FAILED,
pymilvus.exceptions.MilvusException: <MilvusException: (code=2, message=Fail connecting to server on localhost:19530. Timeout)>

**Comment by xiaofan-luan** [comment]
> 

1. please check if you client has connection with your server. telnet localhost:19530 could tell. 
2. if you network is not working, I guess you probably need to check your firewall and maybe check your service is up and running

**Comment by tianduing** [comment]
me too, difficulty problem

**Comment by xiaofan-luan** [comment]
please share the error message if you are looking for solution,
Usually it is caused by:
1. SDK - Server version mismatch.
2. forget to enable use_secure
3. use the wrong username/password

**Comment by pandasar13** [comment]
Hi, I have the same issue. Docker is up and running (I followed all the steps). I am using milvus-haystack

```
from milvus_haystack import MilvusDocumentStore

document_store = MilvusDocumentStore(uri="http://localhost:19530/default")
```

pymilvus.exceptions.MilvusException: <MilvusException: (code=2, message=Fail connecting to server on localhost:19530. Timeout)>

Can somebody help me?

**Comment by jones-so** [comment]
您的邮件创视已经收到，请等待我给您的回复，祝您开心！

**Comment by bardzjhe** [comment]
> MilvusException(Status.CONNECT_FAILED, pymilvus.exceptions.MilvusException: <MilvusException: (code=2, message=Fail connecting to server on localhost:19530. Timeout)>

Same Issue occurs in windows, anyone could help?

**Comment by jones-so** [comment]
您的邮件创视已经收到，请等待我给您的回复，祝您开心！

**Comment by ALLADISAIMANOJ** [comment]
Hey I'm getting this exception. 
MilvusException: <MilvusException: (code=2, message=Fail connecting to server on localhost:19530, illegal connection params or server unavailable)>

It is saying that illegal connection params or server unavailable when I tried to connect using google collab. Can anyone help me ?

**Comment by jones-so** [comment]
您的邮件创视已经收到，请等待我给您的回复，祝您开心！

**Comment by LannyCodes** [comment]
> Hey I'm getting this exception. MilvusException: <MilvusException: (code=2, message=Fail connecting to server on localhost:19530, illegal connection params or server unavailable)>
> 
> It is saying that illegal connection params or server unavailable when I tried to connect using google collab. Can anyone help me ?

Hello, I'm encountering the same error as you. May I ask if you've resolved the issue? And how did you solve it?
-->pymilvus.exceptions.MilvusException: <MilvusException: (code=2, message=Fail connecting to server on 127.0.0.1:19530, illegal connection params or server unavailable)>

**Comment by jones-so** [comment]
您的邮件创视已经收到，请等待我给您的回复，祝您开心！

**Comment by niks-123** [comment]
Hi, I am getting same below error. It was working fine and suddenly started getting this error
Can someone please help resolve this? I am using docker

`pymilvus.exceptions.MilvusException: <MilvusException: (code=2, message=Fail connecting to server on localhost:19530, illegal connection params or server unavailable)>
`

**Comment by gozus19p** [comment]
Same error here: pymilvus.exceptions.MilvusException: <MilvusException: (code=2, message=Fail connecting to server on localhost:19530, illegal connection params or server unavailable)>

It was working fine and then this started to come out.

**Comment by jones-so** [comment]
您的邮件创视已经收到，请等待我给您的回复，祝您开心！

**Comment by gozus19p** [comment]
> Same error here: pymilvus.exceptions.MilvusException: <MilvusException: (code=2, message=Fail connecting to server on localhost:19530, illegal connection params or server unavailable)>
> 
> It was working fine and then this started to come out.

Just a quick note. In my case the problem was the company's proxy server. When I got home, I tried from local network. That problems simply disappear.
I guess that's a problem related with some proxy or firewall configurations.
If that's the case, I would be glad to know how to bypass those, at least for "localhost" hostname. I searched for something like that on Google, but I found nothing.
